src/jdk.jfr/share/classes/jdk/jfr/internal/EventInstrumentation.java

```diff
@@ -248,11 +252,11 @@ private static List<SettingDesc> buildSettingDescs(Class<?> superClass, ClassMod
                             MethodTypeDesc mtd = m.methodTypeSymbol();
                             if ("Z".equals(mtd.returnType().descriptorString())) {
                                 if (mtd.parameterList().size() == 1) {
-                                    ClassDesc paramType = mtd.parameterList().getFirst();
-                                    if (!paramType.isPrimitive()) {
+                                    ClassDesc type = mtd.parameterList().getFirst();
+                                    if (type.isClassOrInterface()) {
```

Comment:
In the original, ASM-based code, there was no such check to ensure the type is non-primitive or non-array. Is this intended?

Reply:
Yes.